javascript/ql/src/semmle/javascript/dataflow/internal/VariableTypeInference.qll

```diff
@@ -223,7 +223,11 @@ abstract class AnalyzedSsaDefinition extends SsaDefinition {
  * Flow analysis for SSA definitions corresponding to `VarDef`s.
  */
 private class AnalyzedExplicitDefinition extends AnalyzedSsaDefinition, SsaExplicitDefinition {
-  override AbstractValue getAnRhsValue() { result = getDef().(AnalyzedVarDef).getAnAssignedValue() }
+  override AbstractValue getAnRhsValue() {
+    result = getDef().(AnalyzedVarDef).getAnAssignedValue()
+    or
+    result = getRhsNode().analyze().getALocalValue()
```

Comment:
`getRhsNode` does not include the default value of a destructuring pattern:
```
result = DataFlow::ssaDefinitionNode(this).getImmediatePredecessor()
```

Since it's not a soundness issue it shouldn't matter a huge amount, but it seems like it would be fairly easy to just use `getAPredecessor` instead.

Reply:
Noted; I'll add that as part of the more general rewrite (if and when I get to it).